### PR TITLE
DOC: Update Series.repeat docstring

### PR DESCRIPTION
**Pull request by pedropazzini**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [ X] PR title is "DOC: update the <your-function-or-method> docstring"
- [ ] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [ X] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ X] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [ X] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```

################################################################################
####################### Docstring (pandas.Series.repeat) #######################
################################################################################

Repeat elements of a Series.

Each element of the Series is repeated 'repeats' times.

Parameters
----------
repeats : int
    The number of repetitions for each element.
*args
    These parameters will be passed to a validation function.
**kwargs
    These parameters will be passed to a validation function.

See also
--------
Series.append : Concatenate two or more Series.
pandas.concat : Concatenate two or more DataFrames.
numpy.repeat : Repeat elements of an array.

Returns
-------
Series
    The repeated version of the Series.

Examples
--------
>>> df = pd.DataFrame({'col1' : ['A','B','C'],'col2' : [ 0 , 1 , 2 ]})
>>> df
col1  col2
0    A     0
1    B     1
2    C     2
>>> df.col1.repeat(3)
0    A
0    A
0    A
1    B
1    B
1    B
2    C
2    C
2    C
Name: col1, dtype: object

################################################################################
################################## Validation ##################################
################################################################################

Errors found:
	Errors in parameters section
		Parameters {'kwargs', 'args'} not documented
		Unknown parameters {'**kwargs', '*args'}
		Parameter "*args" has no type
		Parameter "**kwargs" has no type

```

The errors returned for the script is an error, since t is not ready for **kwargs and *args parameters.

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -918,12 +918,49 @@ def _set_values(self, key, value):
     @deprecate_kwarg(old_arg_name='reps', new_arg_name='repeats')
     def repeat(self, repeats, *args, **kwargs):
         """
-        Repeat elements of an Series. Refer to `numpy.ndarray.repeat`
-        for more information about the `repeats` argument.
+        Repeat elements of a Series.
+
+        Each element of the Series is repeated 'repeats' times.
```
Comment (TomAugspurger) [inline pandas/core/series.py]: Single backtick for parameters like \`repeats\`

**Comment by jorisvandenbossche** [comment]
@pedropazzini Can you check the similar PR https://github.com/pandas-dev/pandas/pull/19985 to see if there is something in there you can use? (eg explanation for the `**kwargs`)

File: pandas/core/series.py [hunk 2]
```diff
@@ -918,12 +918,49 @@ def _set_values(self, key, value):
...
+
+        Parameters
+        ----------
+        repeats : int
+            The number of repetitions for each element.
+        *args
```
Comment (TomAugspurger) [inline pandas/core/series.py]: You can remove args and kwargs. The validation function is an implementation detail that users shouldn't have to worry about. `repeats` is the only valid argument.

You *could* mention, perhaps in `Notes` that "unlike `numpy.repeat`, \`axis\` is not a valid argument".

File: pandas/core/series.py [hunk 3]
```diff
@@ -918,12 +918,49 @@ def _set_values(self, key, value):
...
+        Returns
+        -------
+        Series
+            The repeated version of the Series.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame({'col1' : ['A','B','C'],'col2' : [ 0 , 1 , 2 ]})
+        >>> df
+        col1  col2
```
Comment (TomAugspurger) [inline pandas/core/series.py]: Spacing looks off. Should move it over a few spaces I think. Double check that this is right.

File: pandas/core/series.py (hunk 2 above)
Comment (TomAugspurger) [reply pandas/core/series.py]: Listen to @jorisvandenbossche about args and kwargs, not me :)

Comment (jorisvandenbossche) [reply pandas/core/series.py]: After discussion, you can document them like

```
*args, **kwargs
    Additional keywords have no effect but might be accepted for
    compatibility with numpy.
```

**Comment by jorisvandenbossche** [comment]
@pedropazzini Is there a reason that you closed this?

**Pull request closed without merging**